Libs/MRML/Core/vtkMRMLDiffusionTensorVolumeDisplayNode.cxx

```diff
@@ -471,3 +512,46 @@ int vtkMRMLDiffusionTensorVolumeDisplayNode::GetNthScalarInvariant(int i)
     vtkMRMLDiffusionTensorVolumeDisplayNode::GetSupportedColorModes();
   return modes[i];
 }
+
+//----------------------------------------------------------------------------
+const char *vtkMRMLDiffusionTensorVolumeDisplayNode::getPixelString(double *ijk)
+{
+    vtkImageData *imageData = this->GetVolumeNode()->GetImageData();
+    if(imageData == NULL){
+        return "No Image";
+    }
+
+    vtkIdType point_idx = imageData->FindPoint(ijk[0], ijk[1], ijk[2]);
+
+    if(point_idx == -1)
+      return "Out of bounds";
+
+    if(imageData->GetPointData() == NULL)
+      return "No Point Data";
+
+    if(imageData->GetPointData()->GetTensors() == NULL)
+      return "No Tensor Data";
+
+    double tensor[9];
+    double *a = &tensor[0];
+    imageData->GetPointData()->GetTensors()->GetTuple(point_idx, a);
+
+    int operation;
+
+    operation = this->GetScalarInvariant();
+
+    std::string valueString = this->GetScalarInvariantAsString();
+
+    const float *tensorf = (float*) &tensor[0];
+    this->tensordata->SetTupleValue(0, tensorf);
+    this->tensordata->Modified();
+    this->singlepixelimage->Modified();
+    this->dtimath->SetOperation(operation);
+    this->dtimath->Update();
+    vtkImageData *output = this->dtimath->GetOutput();
+
+    if(output && output->GetNumberOfScalarComponents() > 0)
+        valueString += " " + DoubleToString(output->GetScalarComponentAsDouble(0, 0, 0, 0));
+
+    return valueString.c_str();
```

Comment:
```
const char* foo = getPixelString(...);
// here because valueString is destroyed, foo is pointing to memory that has been destroyed.
```
You might want to return a std::string instead